clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -990,6 +990,12 @@ def warn_main_redefined : Warning<"variable named 'main' with external linkage "
     "has undefined behavior">, InGroup<Main>;
 def ext_main_used : Extension<
     "referring to 'main' within an expression is a Clang extension">, InGroup<Main>;
+def ext_main_invalid_linkage_specification : ExtWarn<
+  "'main' should not be "
+  "'extern \"%select{C|C++}0\"'">, InGroup<Main>;
+def err_main_invalid_linkage_specification : Error<
+  "'main' should not be "
+  "'extern \"%select{C|C++}0\"'">;
```

Comment:
Thinking about this again, I think making these an error by default is a bit too strong; no-one else does that by default; the most you’ll get is a warning from EDG and a pedantic warning from GCC. I think making this a single `ExtWarn` is fine.

Reply:
@Sirraide Thanks for mentioning about the `.Summary`. I’ll keep it in mind for future fixes. I've added the requested changes.

Reply:
> Is it advisable to use force push, or is it better to keep the commit history in PRs?

I don’t think we have a policy on that officially, and it doesn’t really matter because we do squash+merge anyway, so the commit history for a pr can be as messy as it needs to be. *Personally*, however, I’d prefer it if people *didn’t* use force pushes simply because that way I know what was added and what I haven’t reviewed yet instead of having to go over the entire pr again.

Reply:
@Sirraide thanks for the detailed clarification.